resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -63,6 +63,19 @@ private[spark] class Client(
     .map(_.split(","))
     .getOrElse(Array.empty[String])
 
+  // Memory settings
+  private val driverMemory = sparkConf.get("spark.driver.memory", "1g")
+  private val driverMemoryBytes = Utils.byteStringAsBytes(driverMemory)
+  private val driverSubmitServerMemory = sparkConf.get(KUBERNETES_DRIVER_SUBMIT_SERVER_MEMORY)
+  private val driverSubmitServerMemoryBytes = Utils.byteStringAsBytes(driverSubmitServerMemory)
+  private val driverContainerMemoryBytes = driverMemoryBytes + driverSubmitServerMemoryBytes
+  private val memoryOverheadBytes = sparkConf
+    .get(KUBERNETES_DRIVER_MEMORY_OVERHEAD)
+    .map(overhead => Utils.byteStringAsBytes(overhead))
+    .getOrElse(math.max((MEMORY_OVERHEAD_FACTOR * driverContainerMemoryBytes).toInt,
+      MEMORY_OVERHEAD_MIN))
+  private val driverContainerMemoryWithOverhead = driverContainerMemoryBytes + memoryOverheadBytes
+
```

Comment:
I'd rather leave `config.scala` for just the config values and keys - there should be no logic there.